src/borg/archiver/check_cmd.py

```diff
@@ -180,11 +178,13 @@ def build_parser_check(self, subparsers, common_parser, mid_common_parser):
         Consequently, if lost chunks were repaired earlier, it is advised to run
         ``--repair`` a second time after creating some new backups.
 
-        If ``--repair --undelete-archives`` is given, Borg will scan the repository
+        If ``--repair --find-lost-archives`` is given, Borg will scan the repository
         for archive metadata and if it finds some where no corresponding archives
-        directory entry exists, it will create the entries. This is basically undoing
-        ``borg delete archive`` or ``borg prune ...`` commands and only possible before
-        ``borg compact`` would remove the archives' data completely.
+        directory entry exists (neither a normal not-deleted archive entry nor a
+        soft-deleted archive entry), it will create the entries as not-deleted.
+        This will make archives reappear for which the directory entry was lost.
+        This is only possible before ``borg compact`` would remove the archives'
+        data completely.
```

Comment:
Since `--find-lost-archives` is no longer limited to `--repair`, this should be explained in the general (non-repair mode) section.

How about adding this to the third paragraph (no. 2 in the list of `check`' steps) after the third sentence (i.e. after "This requires reading archive and file metadata, but not data."), also updating the following sentence about `--verify-data`:

```
To scan for archives whose entries were lost from the archive directory, pass
`--find-lost-archives``. It requires reading all data and is hence very time
consuming. To additionally cryptographically verify the file (content) data
integrity pass ``--verify-data``, which is hence even more time consuming.
```

Then how about adding the following as last paragraph of the general section (i.e. after the paragraph about `--verify-data`):

```
Similar to ``--verify-data``, the ``--find-lost-archives`` option will also
scan the whole repository, but tells Borg to search for lost archive metadata.
If Borg encounters any archive metadata that doesn't match with an
existing archive entry (both normal/not-deleted archives, and soft-deleted
archives), it means that an archive entry was lost from the directory. Unless
``borg compact`` is called, these archives can be fully restored with
``--repair``. Please note that ``--find-lost-archives`` must read a lot of
data from the repository and is thus very time consuming. You can not use
``--find-lost-archives`` with ``--repository-only``.
```

The paragraph in the repair mode section might then be shortened to the following:

```suggestion
       As explained earlier, ``--find-lost-archives`` causes Borg to scan the
       repository for dangling archive metadata. Together with ``--repair``
       Borg will use that archive metadata to re-create entries in the
       directory. This will make previously lost archives reappear as
       normal/not-deleted archive. Note that this is only possible as long as
       you didn't run ``borg compact``, because ``borg compact`` would remove
       the archives' data completely.
```

Reply:
Hmm, applying this many/big changes from a suggestion via the github UI does not work.

I'll fetch the stuff to local repo and will do some updates...

Reply:
fixed.